lib/net.js

```diff
@@ -1135,6 +1135,7 @@ Server.prototype._listen2 = function(address, port, addressType, backlog, fd) {
     debug('_listen2: create a handle');
     var rval = createServerHandle(address, port, addressType, fd);
     if (util.isNumber(rval)) {
+      address = address || '0.0.0.0';
```

Comment:
I didn't notice before that this is just for formatting an exception. I don't know how I feel about setting the value artificially just for that, but I'm OK with the rest of the PR.